pkg/backup/resource_backupper.go

```diff
@@ -174,37 +176,57 @@ func (rb *defaultResourceBackupper) backupResource(
 	)
 
 	var namespacesToList []string
-	if resource.Namespaced {
-		namespacesToList = getNamespacesToList(rb.namespaces)
-	} else {
-		namespacesToList = []string{""}
-	}
-	for _, namespace := range namespacesToList {
-		resourceClient, err := rb.dynamicFactory.ClientForGroupVersionResource(gv, resource, namespace)
+	// If processing the Namespace resource specifically and you've specified to include only some,
+	// fetch them individually instead of a full list of all in the cluster:
+	if resource.Kind == "Namespace" && !rb.namespaces.IncludeEverything() {
```

Comment:
The `!rb.namespaces.IncludeEverything()` check isn't going to work here. This expression will evaluate to true if you have any excludes, even if you don't have any specific includes, for example. You're better off checking if length of `GetIncludes()` >= 1 and it doesn't contain `*`. Or we could create another helper `HasSpecificIncludes` or something like that.

Reply:
Updated with another pass at that, I see validation that if * is in the list it must be the only element in the list, so I hope I got it right this time.